etl/steps/data/garden/demography/2023-03-31/population/__init__.py

```diff
@@ -212,20 +222,50 @@ def add_world(df: Table) -> Table:
     return df
 
 
-def add_historical_regions(df: Table) -> Table:
+def add_historical_regions(df: Table, tb_regions: Table) -> Table:
     """Add historical regions.
 
-    Systema Globalis from Gapminder contains historical regions. We add them to the data. These include
-    Yugoslavia, USSR, etc.
+    Historical regions are added using different techniques:
 
-    Note that this is added after regions and world regions have been obtained, to avoid double counting.
+    1. Systema Globalis from Gapminder contains historical regions. We add them to the data. These include
+    Yugoslavia, USSR, etc. Note that this is added after regions and world regions have been obtained, to avoid double counting.
+    2. Add historical regions by grouping and summing current countries.
     """
+
+    # 1. Add from Systema Globalis
     log.info("population: loading data (Gapminder Systema Globalis)")
     gapminder_sg = load_gapminder_sys_glob_former()
     # map source name
     gapminder_sg["source"] = SOURCES_NAMES["gapminder_sg"]
-
-    return pr.concat([df, gapminder_sg], ignore_index=True)
+    # Add to main table
+    tb = pr.concat([df, gapminder_sg], ignore_index=True)
+
+    # 2. Add historical regions by grouping and summing current countries.
+    for ccode in FORMER_COUNTRIES_CODES:
+        # Get former country name and end year (dissolution)
+        former_country_name = tb_regions.loc[ccode, "name"]
+        end_year = tb_regions.loc[ccode, "end_year"]
+        # Sanity check
+        assert former_country_name not in set(
+            tb["country"]
+        ), f"{former_country_name} already in table (either import it via Systema Globalis or manual aggregation)!"
+        # Get list of country successors (equivalent of former state nowadays) and end year (dissolution of former state)
+        ccodes_successors = ast.literal_eval(tb_regions.loc[ccode, "successors"])
+        successor_names = tb_regions.loc[ccodes_successors, "name"].tolist()
+        # Filter table accordingly
+        tb_ = tb[(tb["year"] <= end_year) & (tb["country"].isin(successor_names))]
+        # Filter rows (only preserve years where all countries have data)
+        year_filter = tb_.groupby("year")["country"].nunique() == len(successor_names)
+        year_filter = year_filter[year_filter].index.tolist()
+        tb_ = tb_[tb_["year"].isin(year_filter)]
+        # Perform operations
+        tb_ = tb_.groupby("year", as_index=False).agg(
+            {"population": sum, "source": lambda x: "; ".join(sorted(set(x)))}
+        )
+        tb_["country"] = former_country_name
+        # Add to main table
+        tb = pr.concat([tb, tb_], ignore_index=True)
+    return tb
```

Comment:
The result is roughly consistent with the population from [wikipedia](https://en.wikipedia.org/wiki/Demographics_of_the_Soviet_Union#Population).